### PR TITLE
Automated Integration Testing on Dot net Client

### PR DESCRIPTION
**Pull request by mahajanadhitya**

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -109,30 +109,30 @@ blocks:
...
+    dependencies: [ ]
+    task:
+      agent:
+        machine:
+          type: s1-prod-ubuntu20-04-amd64-2
+      jobs:
+        - name: 'Build and test'
+          commands:
+            - cd test/docker && docker-compose up -d && cd ../..
+            - export onSemaphore='set'
```
Comment (anchitj) [inline .semaphore/semaphore.yml:122]: Better to rename it something like `SEMAPHORE_SKIP_FLAKY_TETSTS
```suggestion
            + export SEMAPHORE_SKIP_FLAKY_TETSTS='true'
```

File: test/Confluent.Kafka.IntegrationTests/Tests/AdminClient_AlterConfigs.cs [hunk 2]
```diff
@@ -34,6 +34,10 @@ public partial class Tests
         public void AdminClient_AlterConfigs(string bootstrapServers)
         {
             LogToFile("start AdminClient_AlterConfigs");
+            string onSemaphore = Environment.GetEnvironmentVariable("onSemaphore");
+            if(onSemaphore!=null){
+                return;
+            }
```
Comment (anchitj) [inline test/Confluent.Kafka.IntegrationTests/Tests/AdminClient_AlterConfigs.cs:40]: We can move the `Environment.GetEnvironmentVariable` functionality into the `Tests.cs` file and create a static function there called `SkipSemaphoreFlakyTests`. Individual tests can use this function to decide whether to skip a test. 

Additionally, let's log using `LogToFile` whenever we skip a flaky test.

File: test/Confluent.SchemaRegistry.IntegrationTests/Tests/ProtobufWithReferences.cs [hunk 3]
```diff
@@ -25,6 +25,11 @@ public static partial class Tests
         [Theory, MemberData(nameof(SchemaRegistryParameters))]
```
Comment (anchitj) [inline test/Confluent.SchemaRegistry.IntegrationTests/Tests/ProtobufWithReferences.cs:25]: This test looks to be failing always. Can you add a `FIXME` comment here to fix this test?
And let's comment this line `cd test/Confluent.SchemaRegistry.IntegrationTests && dotnet test -l "console;verbosity=normal" && cd ../..` in the semaphore.yml.

Comment (mahajanadhitya) [reply test/Confluent.SchemaRegistry.IntegrationTests/Tests/ProtobufWithReferences.cs:25]: Done !

File: test/Confluent.SchemaRegistry.IntegrationTests/Tests/ProtobufWithReferences.cs [hunk 4]
```diff
@@ -25,6 +25,12 @@ public static partial class Tests
...
         public static void ProtobufWithReferences(Config config)
         {
+            // FIXME : The Tests always Fails !!
+            bool skipFlakyTests = semaphoreSkipFlakyTests();
+            if (skipFlakyTests)
+            {
+                return;
+            }
```
Comment (anchitj) [inline test/Confluent.SchemaRegistry.IntegrationTests/Tests/ProtobufWithReferences.cs:33]: This is not needed now as we aren't running this test suite. Can be removed.

**Review by anchitj — APPROVED** [review]

**Comment by anchitj** [comment]
LGTM! Just need to do the nit change before merging.

**Pull request merged**